### ISSUE TITLE
[netbox.netbox.netbox_cable] More than one result returned for termination_a in case of VC or switch stack.

### ISSUE THREAD

**Issue by igloo777** [netbox-community/ansible_modules#465]
Hi all! 

Collection version: 3.0.0
Netbox version: 2.10.4

I created stack (VC) with 2 members in Netbox. 
If I create cable link between master and non-master devices on Web page - it works correctly.
But if I do the same with ansible, I get error `More than one result returned for termination_a` or `More than one result returned for termination_b` - depends at which side I ctreate link (master or non-master).

Seems like it happens because the master switch obtain and show the full port range of both stack members, but do not change the member id in the port name. So, foMay ber sample, even in web interface we have two (or even more - depends on number of stack members) Gi1/0/1 ports. 

I can't reproduce such issue with a standalone switches. 

I'm not sure if I would create an issue for Netbox developers to fix port-naming logic, but I will try.

**Comment by igloo777** [comment]
Hmm... I had try.. )
https://github.com/netbox-community/netbox/issues/5944

**Comment by FragmentedPacket** [comment]
Do you mind providing the play/task for this with any sensitive data removed? And maybe an example of what NetBox data you have for these two?

**Comment by jefvantongerloo** [comment]
Related: https://github.com/netbox-community/ansible_modules/issues/402

**Comment by FragmentedPacket** [comment]
That's a good call out @jefvantongerloo.

@igloo777 If you use the PR that is in to fix that, does this issue go away?

**Comment by igloo777** [comment]
> That's a good call out @jefvantongerloo.
> 
> @igloo777 If you use the PR that is in to fix that, does this issue go away?

PR fixed the issue with more than 2 members in VC, but the issue with cables still here..

**Comment by igloo777** [comment]
> Do you mind providing the play/task for this with any sensitive data removed? And maybe an example of what NetBox data you have for these two?

Code
```
---
- hosts: localhost
  collections: netbox.netbox
  vars:
    netbox_url: "https://localhost"
    netbox_token: "token"

  tasks:
    - name: Configure test1
      netbox.netbox.netbox_device:
        netbox_url: "{{ netbox_url }}"
        netbox_token: "{{ netbox_token }}"
        data:
          name: test1
          device_type: PowerSwitch N1548P
          device_role: Access Switch
          platform: Dell OS6
          site: site1
          status: planned
        state: present
        validate_certs: false

    - name: Configure test-stack
      netbox.netbox.netbox_virtual_chassis:
        netbox_url: "{{ netbox_url }}"
        netbox_token: "{{ netbox_token }}"
        data:
          name: test-stack
          master: test1
        state: present
        validate_certs: false

    - name: Configure test2
      netbox.netbox.netbox_device:
        netbox_url: "{{ netbox_url }}"
        netbox_token: "{{ netbox_token }}"
        data:
          name: test2
          device_type: PowerSwitch N1548P
          device_role: Access Switch
          platform: Dell OS6
          site: site1
          status: planned
          virtual_chassis: test-stack
          vc_position: 2
        state: present
        validate_certs: false

    - name: Cables
      netbox.netbox.netbox_cable:
        netbox_url: "{{ netbox_url }}"
        netbox_token: "{{ netbox_token }}"
        data:
          termination_a_type: "dcim.interface"
          termination_a:
            device: test1
            name: Te1/0/1
          termination_b_type: "dcim.interface"
          termination_b:
            device: test2
            name: Te1/0/1
          type: dac-passive
          status: planned
        state: present
        validate_certs: false
```

Log:
```
The full traceback is:
  File "/tmp/ansible_netbox.netbox.netbox_cable_payload_81sv4blp/ansible_netbox.netbox.netbox_cable_payload.zip/ansible_collections/netbox/netbox/plugins/module_utils/netbox_utils.py", line 532, in _nb_endpoint_get
    response = nb_endpoint.get(**query_params)
  File "/home/igloo/_git/robo/local/ansible/playbooks/venv/lib/python3.8/site-packages/pynetbox/core/endpoint.py", line 141, in get
    raise ValueError(
fatal: [localhost]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "data": {
                "color": null,
                "label": null,
                "length": null,
                "length_unit": null,
                "status": "planned",
                "tags": null,
                "termination_a": {
                    "device": "test1",
                    "name": "Te1/0/1"
                },
                "termination_a_type": "dcim.interface",
                "termination_b": {
                    "device": "test2",
                    "name": "Te1/0/1"
                },
                "termination_b_type": "dcim.interface",
                "type": "dac-passive"
            },
            "netbox_token": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "netbox_url": "https://localhost",
            "query_params": null,
            "state": "present",
            "validate_certs": false
        }
    }
}

MSG:

More than one result returned for termination_a
```

**Comment by rodvand** [comment]
I have investigated this and understand the issue a bit more. The problem is currently how NetBox returns interfaces when you search for interfaces on a device which is also defined as the master in a virtual chassis.

An alternative, if you are able to query and get the interface ID somehow is to define the ID of the interface directly in the playbook:

```
termination_a:
  id: <ID>
```

As NetBox currently have no way of returning all the interfaces related to a single device in a virtual chassis (when the device is master of the virtual chassis) I'm closing this as won't fix. I created a discussion on the NetBox repository https://github.com/netbox-community/netbox/discussions/8679 to see if this is the desired behaviour. 

If anything changes with how we can query the interface we can re-open and maybe make some adjustments.
